backend/database/migrations/2022_08_11_112450_create_companies_table.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+use Internships\Enums\CompanyStatus;
+use Internships\Models\User;
+
+return new class() extends Migration {
+    public function up(): void
+    {
+        Schema::create("companies", function (Blueprint $table): void {
+            $table->id();
+            $table->string("name");
+            $table->text("description");
+            $table->foreignIdFor(User::class)->constrained()->cascadeOnDelete();
+            $table->json("address");
```

Comment:
Are you think that json field will be good? What about searching via city, and so on? The same with `contact_details` column.
I would like to keep these data as separated columns.

Reply:
Laravel has options to serach in JSON fields

Reply:
> Are you think that json field will be good? What about searching via city, and so on? The same with `contact_details` column. I would like to keep these data as separated columns.

We use the laravel-castable-data-transfer-object package, which allows us to cast JSON columns to objects. In addition as above Laravel allows us to conveniently search in JSON fields: https://laravel.com/docs/9.x/queries#json-where-clauses

Reply:
Ok, then I don't care. But in my opinion these fields should be extracted and put as separated columns.